### ISSUE TITLE
Android WebView lacks onload handler due to rubicon-java inability to subclass

### ISSUE THREAD

**Issue by paulproteus** [beeware/toga#1020]
**Describe the bug**
Currently, Android WebView lacks an onload handler.

**To Reproduce**
1. Create a Toga app that uses WebView
2. Add an onload handler (as the [webview demo does!](https://github.com/beeware/toga/blob/master/examples/webview/webview/app.py))

**Expected behavior**

Expect the onload handler to be fired when the page finishes loading.

**Rationale**

Android's WebView implementation needs an instance the `WebViewClient` class, or an instance of a `WebViewClient` subclass, in order to send notifications of pages completing loading.

rubicon-java doesn't support subclassing Java classes.

If it did, the code might look like:

```python
from ..libs import android_widgets

class TogaWebViewClient(android_widgets.WebViewClient):
    def __init__(self, on_load=None):
        super().__init__()
        self._on_load = on_load

    class onPageFinished(self, _webview, _url):
        if self._on_load:
            self._on_load()
```

This depends on https://github.com/beeware/rubicon-java/issues/25 (or something like it).

**Comment by sarthaksharma058439** [comment]
Check this video .....he make calculator in beeware and deploy to android device

https://youtu.be/RllR6FRdG6w

**Comment by xloem** [comment]
The blocking issue here is a big project and it's appropriate to make helper java classes to work around issues until the blocking issue is resolved.

**Comment by xloem** [comment]
One possible solution to issues like this would be to add wrappers for all the android api's abstract base classes to the [android template](https://github.com/beeware/briefcase-android-gradle-template/tree/3.8/%7B%7B%20cookiecutter.formal_name%20%7D%7D/app/src/main).  The wrappers would take java interfaces in their constructors and pass off their abstract method calls to the interfaces.  This solution could be implemented by somebody who didn't know the project well.

**Comment by freakboy3742** [comment]
@xloem Yes, this is a possible approach; however, it's an approach that I'd rather avoid. Baking wrappers for specific classes into the Android template would make the template GUI toolkit specific - you would only be able to extend the classes that we've pre-declared. Consider the case of wrapping a third party Android library, or an Android base library that Toga doesn't require.

I would be more inclined to support a metadata-based approach which allows toga-android to provide/include a jar file that contains class implementations that can be wrapped, and metadata in the toga-android wheel/egg-info that briefcase can use to obtain/install that jar file as needed in a generic Android template.

Better still would be a mechanism for that lets the Android template, or a template post-processing step, roll out the base classes based on a metadata declaration in the toga-android package. Post-processing is a feature that has been requested in other contexts (e.g., to do asset processing), so this could be a natural fit.

**Comment by xloem** [comment]
Thanks for your response.  In Kivy their solution was to expose in a variable the android java source path, so files could be copied into it during build.  Natural fits are always good though, you want as many birds with your stones as you can get =)

**Comment by t-arn** [comment]
@freakboy3742 This issue can be closed now
